### PR TITLE
Arm LED Subsystem

### PR DESCRIPTION
**Pull request by kirbt**
Subsystem for managing arm led by sending http get requests

### CODE REVIEW

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+		//
+
+		enabled = 1;
+		effectIndex = 0;
+		color1 = ColorSelector.RED;
+		color2 = ColorSelector.BLACK;
+		color3 = ColorSelector.RED;
+
+		//
+		client =
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:84]: We should wrap these http calls in a try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+		setLEDTeleop();
+	}
+
+	// METHODS
+
+	/** Called during autonomous to be green (color might be changed later?) */
+	public void setLEDAutonomous() {
+		color1 = ColorSelector.GREEN;
+		request = HttpRequest.newBuilder().uri(URI.create(getURI())).build();
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:137]: Try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+		updateLED();
+	}
+
+	/** Called during Teleop to set LED to red or blue based off alliance. */
+	public void setLEDTeleop() {
+
+		if (DriverStation.getAlliance() == DriverStation.Alliance.Red) {
+			// red
+			color1 = ColorSelector.RED;
+			request = HttpRequest.newBuilder().uri(URI.create(getURI())).build();
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:147]: Try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+		} else {
+			// blue
+			color1 = ColorSelector.BLUE;
+			request = HttpRequest.newBuilder().uri(URI.create(getURI())).build();
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:152]: Try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+	}
+
+	/** Potentially called during Teleop after alliance selection to represent overall team colors */
+	public void setLEDAlliance() {
+		effectIndex = 1;
+
+		color1 = color1Chooser.getSelected();
+		color2 = color2Chooser.getSelected();
+
+		request = HttpRequest.newBuilder().uri(URI.create(getURI())).build();
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:164]: Try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+	}
+
+	/*
+	 * Updates the LED by sending an HTTP request
+	 */
+	public void updateLED() {
+		try {
+			client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
+		} catch (Exception e) {
+			System.out.println("Failed to http request");
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:175]: Could add error message to output so we know what to fix

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+		request = HttpRequest.newBuilder().uri(URI.create(getURI())).build();
+		updateLED();
+	}
+
+	/*
+	 * Updates the LED by sending an HTTP request
+	 */
+	public void updateLED() {
+		try {
+			client.sendAsync(request, HttpResponse.BodyHandlers.ofString());
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:173]: Try catch

File: src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+			return;
+		}
+		enabled = 0;
+
+		updateLED();
+	}
+
+	// TODO: finish URI getter
+	public String getURI() {
+		return "https://"
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/ArmLEDSubsystem.java:195]: Is this really https out just http?

**Review by bobtseattle — CHANGES_REQUESTED** [review]
We need to make sure we aren't crashing robot any where if http lib has a problem.

**Review by bobtseattle — APPROVED** [review]

**Comment by obj-obj** [comment]
image

**Comment by obj-obj** [comment]
This will truly be one of the merges of all time

**Pull request merged**